xknx/cemi/cemi_frame.py

```diff
@@ -195,58 +220,459 @@ def from_knx_data_link_layer(cemi: bytes, code: CEMIMessageCode) -> CEMIFrame:
+
+    def __init__(self, *, property_info: CEMIMPropInfo):
+        """Initialize CEMIMPropReadRequest object."""
+        self.property_info = property_info
+
+    def calculated_length(self) -> int:
+        """Get length of CEMI data."""
+        return self.property_info.calculated_length()
+
+    def to_knx(self) -> bytes:
+        """Serialize to CEMI raw data."""
+        return self.property_info.to_knx()
+
+    @classmethod
+    def from_knx(cls, raw: bytes) -> CEMIData:
+        """Parse/deserialize from KNX/IP raw data."""
+        return cls(property_info=CEMIMPropInfo.from_knx(raw))
+
+    def __repr__(self) -> str:
+        """Return object as readable string."""
+        return f"CEMIMPropReadRequest({self.property_info._property_repr()})"
+
+
+class CEMIMPropReadResponse(CEMIData):
+    """Representation of CEMI Device Management Property Read Response."""
+
+    def __init__(
+        self,
+        *,
+        property_info: CEMIMPropInfo,
+        data: bytes,
+    ):
+        """Initialize CEMIMPropReadResponse object."""
+        self.property_info = property_info
+        self.data = data
+
+    @property
+    def error_code(self) -> CEMIErrorCode | None:
+        """Return an optional CEMI error code."""
+        if self.property_info.number_of_elements == 0:
+            return CEMIErrorCode(self.data[0])
+
+        return None
+
+    def calculated_length(self) -> int:
+        """Get length of CEMI data."""
+        return CEMIMPropInfo.LENGTH + len(self.data)
+
+    def to_knx(self) -> bytes:
+        """Serialize to CEMI raw data."""
+        return self.property_info.to_knx() + self.data
+
+    @classmethod
+    def from_knx(cls, raw: bytes) -> CEMIData:
+        """Parse/deserialize from KNX/IP raw data."""
+        if len(raw) < 7:
+            raise CouldNotParseCEMI(
+                f"CEMI Property Read Response too small. Length: {len(raw)}; CEMI: {raw.hex()}"
             )
-            + self.flags.to_bytes(2, "big")
-            + self.src_addr.to_knx()
-            + self.dst_addr.to_knx()
-            + npdu_len.to_bytes(1, "big")
-            + tpdu
+
+        property_info = CEMIMPropInfo.from_knx(raw[0 : CEMIMPropInfo.LENGTH])
+
+        # Did we get an error?
+        if property_info.number_of_elements == 0 and len(raw) != 7:
+            raise CouldNotParseCEMI(
+                f"Invalid CEMI error response length: {len(raw)}; CEMI: {raw.hex()}"
+            )
+
+        return cls(property_info=property_info, data=raw[CEMIMPropInfo.LENGTH :])
+
+    def __repr__(self) -> str:
+        """Return object as readable string."""
+        _data = (
+            f'data="{self.data.hex()}" '
+            if self.property_info.number_of_elements != 0
+            else ""
+        )
+        return (
+            f"CEMIMPropReadResponse("
+            f"{self.property_info._property_repr()}"
+            f'error_code="{self.error_code}" '
+            f"{_data} )"
```

Comment:
```suggestion
            f"{_data})"
```
`_data` already has ends with a space